config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
```

Comment:
Tomorrow I'll take a shot at versioning snapshots and making things die for log entries that aren't understood - that'll make things more robust.

Still not sure what a great deprecation story is for when it's safe to delete code for handling old versions because of the uncertainty of snapshot timing. We could do something simple like run a cleanup goroutine off a snapshot request 15 minutes after a to-be-deprecated log entry gets committed (or if we see one in the log at startup).

This would only cause up to 4 snapshots per hour (or whatever) even if there were tons of these deprecated entries flying by, and would eliminate the deprecated entries within 15 minutes of finishing an upgrade once there were no servers producing the old entries any longer. The ongoing cost would be very low as long as the cleanup routine doesn't need to scan the logs; it can just select on a one-item buffered channel that gets pinged in a non-blocking fashion whenever we encounter something that needs cleanup.